### PR TITLE
remove datetime tests with pint

### LINKED ISSUE #3778: Two strange errors in tests TestVariable using .copy()
Setup is as described in #3777.

The two failures are the following ones:
```
____________________ TestVariable.test_index_0d_not_a_time _____________________

self = <xarray.tests.test_units.TestVariable object at 0x7f3926b2ad00>

    def test_index_0d_not_a_time(self):
        d = np.datetime64("NaT", "ns")
        x = self.cls(["x"], [d])
>       self._assertIndexedLikeNDArray(x, d)

xarray/tests/test_variable.py:206: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <xarray.tests.test_units.TestVariable object at 0x7f3926b2ad00>
variable = <xarray.Variable (x: 1)>
<Quantity(['NaT'], 'meter')>
expected_value0 = numpy.datetime64('NaT'), expected_dtype = None

    def _assertIndexedLikeNDArray(self, variable, expected_value0, expected_dtype=None):
        """Given a 1-dimensional variable, verify that the variable is indexed
        like a numpy.ndarray.
        """
        assert variable[0].shape == ()
        assert variable[0].ndim == 0
        assert variable[0].size == 1
        # test identity
>       assert variable.equals(variable.copy())
E       AssertionError: assert False
E        +  where False = <bound method Variable.equals of <xarray.Variable (x: 1)>\n<Quantity(['NaT'], 'meter')>>(<xarray.Variable (x: 1)>\n<Quantity(['NaT'], 'meter')>)
E        +    where <bound method Variable.equals of <xarray.Variable (x: 1)>\n<Quantity(['NaT'], 'meter')>> = <xarray.Variable (x: 1)>\n<Quantity(['NaT'], 'meter')>.equals
E        +    and   <xarray.Variable (x: 1)>\n<Quantity(['NaT'], 'meter')> = <bound method Variable.copy of <xarray.Variable (x: 1)>\n<Quantity(['NaT'], 'meter')>>()
E        +      where <bound method Variable.copy of <xarray.Variable (x: 1)>\n<Quantity(['NaT'], 'meter')>> = <xarray.Variable (x: 1)>\n<Quantity(['NaT'], 'meter')>.copy

xarray/tests/test_variable.py:151: AssertionError
_____________________ TestVariable.test_equals_all_dtypes ______________________

self = <xarray.tests.test_units.TestVariable object at 0x7f39282b4e50>

    def test_equals_all_dtypes(self):
        for v, _ in self.example_1d_objects():
            v2 = v.copy()
>           assert v.equals(v2)
E           AssertionError: assert False
E            +  where False = <bound method Variable.equals of <xarray.Variable (x: 3)>\n<Quantity(['2000-01-01T00:00:00.000000000' '2000-01-02T00:00:00.000000000'\n '2000-01-03T00:00:00.000000000'], 'meter')>>(<xarray.Variable (x: 3)>\n<Quantity(['2000-01-01T00:00:00.000000000' '2000-01-02T00:00:00.000000000'\n '2000-01-03T00:00:00.000000000'], 'meter')>)
E            +    where <bound method Variable.equals of <xarray.Variable (x: 3)>\n<Quantity(['2000-01-01T00:00:00.000000000' '2000-01-02T00:00:00.000000000'\n '2000-01-03T00:00:00.000000000'], 'meter')>> = <xarray.Variable (x: 3)>\n<Quantity(['2000-01-01T00:00:00.000000000' '2000-01-02T00:00:00.000000000'\n '2000-01-03T00:00:00.000000000'], 'meter')>.equals

xarray/tests/test_variable.py:386: AssertionError
```

I have no idea what’s wrong here.

### PR DESCRIPTION
**Pull request by keewis**
Since it does not make sense to try to wrap `datetime` objects in `pint`, I'm removing those tests. They still exposed a bug in the compatibility code of `isnat`: if `dask` is installed, the code would always use `pd.isnull`.

 - [x] Closes #3778
 - [ ] Tests added
 - [x] Passes `isort -rc . && black . && mypy . && flake8`
 - [x] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API

### CODE REVIEW

File: xarray/core/duck_array_ops.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def fail_on_dask_array_input(values, msg=None, func_name=None):
 
 
 if hasattr(np, "isnat") and (
-    dask_array is None or hasattr(dask_array_type, "__array_ufunc__")
+    dask_array_type == () or hasattr(dask_array_type[0], "__array_ufunc__")
```
Comment (keewis) [inline xarray/core/duck_array_ops.py]: not sure about this, does anyone have a better idea?

**Comment by max-sixty** [comment]
LGTM!

File: xarray/core/duck_array_ops.py (hunk 1 above)
Comment (dcherian) [reply xarray/core/duck_array_ops.py]: can you describe the problem please?

Comment (keewis) [reply xarray/core/duck_array_ops.py]: the issue is that `dask_array_type` is always a tuple: https://github.com/pydata/xarray/blob/3ef75aeffaf89aa1c7ea9957e5bdf8d1b36aac54/xarray/core/pycompat.py#L5-L11
so for checking if `dask.array.Array` implements `__array_function__` we need to use its first element

Comment (dcherian) [reply xarray/core/duck_array_ops.py]: wait we are now numpy >= 0.15 and dask >= 2.2. Can we get rid of the if condition?

Comment (keewis) [reply xarray/core/duck_array_ops.py]: I think so? `__array_ufunc__` is supported since `numpy>=0.13` and `dask>=0.15`.

Comment (keewis) [reply xarray/core/duck_array_ops.py]: I replaced the condition with a call to `_dask_or_eager_func`.

**Pull request merged**